scio-jdbc/src/test/scala/com/spotify/scio/jdbc/sharded/ShardedQueryTests.scala

```diff
@@ -52,7 +52,7 @@ class ShardedQueryTests extends AnyFlatSpec {
   "toSelectStatement" must "produce the correct statement for hex string in upper case" in {
 
     val shardQuery = RangeShardQuery[HexUpperString](
-      Range(HexUpperString("a"), HexUpperString("f")),
+      Range(HexUpperString("A"), HexUpperString("F")),
```

Comment:
Looks like `HexUpperString` etc. does to upper/lower and can handle such cases. Should they also validate that prefix is valid hex & fail fast? We can do that as a separate PR.

Reply:
That was one my changes. I removed that logic from `HexUpperString` etc. completely and moved it to `RangeShardStringCodec`s to keep it one place. It was confusing to have in both `HesUpper/LowerStrings` and `ShardCodec`s. I think adding the validation logic is a good idea even though it isn't going to give 100% guarantee that there are no bad strings in the table. We would apply validation only to min and max strings and there might be some badly formatted strings in between.